gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -237,6 +233,16 @@ public void run() {
         LOG.error("Failed to close all open resources", t);
       }
 
+      for (Map.Entry<Optional<Fork>, Optional<Future<?>>> forkAndFuture : this.forks.entrySet()) {
```

Comment:
Instead of canceling all forks if one fork fails, can you let all forks finish, record the failed forks and when retrying the task, only re-run the failed forks?